tests/test_batch.py

```diff
@@ -959,3 +961,150 @@ def test_sequence_builder_field_is_accessible(
             mock_request.last_request.url
             == f"{self.sdk._client.endpoints.core}/api/v2/batches/{first_batch.id}"
         )
+
+    @pytest.mark.parametrize(
+        ("exception", "expected_exception"),
+        [
+            (requests.Timeout, requests.Timeout),
+            (
+                lambda: requests.HTTPError(
+                    "500 Server Error", response=mock_500_http_error_response()
+                ),
+                BatchFetchingError,
+            ),
+            (requests.ConnectionError("Connection refused"), requests.ConnectionError),
+        ],
+        ids=["timeout", "http_500", "connection_error"],
+    )
+    def test_create_batch_retries_on_transient_errors(
+        self,
+        mock_request,
+        exception,
+        expected_exception,
+    ):
+        """Test that GET batch call retries on transient errors
+        after we create a batch."""
+        mock_request.reset_mock()
+
+        # Register the URI with the appropriate exception
+        exc = exception() if callable(exception) else exception
+        mock_request.register_uri(
+            "GET",
+            f"https://apis.pasqal.cloud/core-fast/api/v2/batches/{self.batch_id}",
+            exc=exc,
+        )
+
+        with contextlib.suppress(expected_exception):
+            self.sdk.create_batch(
+                serialized_sequence=self.pulser_sequence,
+                jobs=[self.simple_job_args],
+                device_type=DeviceTypeName.EMU_MPS,
+                wait=True,
+            )
+
+        # Assertions to verify retry behavior
+        # There is one call to create the batch, then 6 calls
+        # to get the batch, 5 of which are retries
+        assert mock_request.call_count == 7
+        assert mock_request.last_request.method == "GET"
+        assert (
+            mock_request.last_request.path
+            == f"/core-fast/api/v2/batches/{self.batch_id}"
+        )
+        assert mock_request.last_request.matcher.call_count == 6
+
+    @pytest.mark.parametrize(
+        ("exception", "expected_exception"),
+        [
+            (requests.Timeout, requests.Timeout),
```

Comment:
So this is not directly related to your MR but I think it would be better if we raised BatchFetchingError for all types of network related errors as well. THis would make it easier for users to understand which step went wrong